addons/mail/models/mail_alias_domain.py

```diff
@@ -128,7 +134,9 @@ def create(self, vals_list):
             if vals.get('catchall_alias'):
                 vals['catchall_alias'] = self.env['mail.alias']._sanitize_alias_name(vals['catchall_alias'])
             if vals.get('default_from'):
-                vals['default_from'] = self.env['mail.alias']._sanitize_alias_name(vals['default_from'])
+                vals['default_from'] = self.env['mail.alias']._sanitize_alias_name(
+                    vals['default_from'], is_email=True
+                )
```

Comment:
the code of the create / write is redundant, can we make a method that sanitize the values ? (take a dictionnary, the write method just call it, the create iterate and call it in a loop)